python_modules/dagster/dagster/_core/definitions/auto_materialize_rule.py

```diff
@@ -119,7 +122,8 @@ class AutoMaterializeRuleEvaluation(NamedTuple):
     evaluation_data: Optional[AutoMaterializeRuleEvaluationData]
 
 
-class RuleEvaluationContext(NamedTuple):
+@dataclass
+class RuleEvaluationContext:
```

Comment:
turned this into a dataclass so that I could use cached_property on it.

Reply:
We should in general use `frozen=True` for dataclasses I believe